### PR TITLE
Update DG Draft with Edit Role Implementation

### LINKED ISSUE #150: Update implementation of Edit role feature  in UG

### PR DESCRIPTION
**Pull request by BrendonLau**
Edit Role implementation consists of:
1. a UML class diagram to describe how the classes are implemented
2. Sequence diagram showing how the classes interact with one another

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -105,6 +105,35 @@ The `JsonAdaptedCompany` also contains a list of roles in `List<JsonAdaptedRole>
...
 
 ---
+## **Implementation**
+
+### Edit role feature
+The `editRole` command for the `Role` item allows the user to update any fields by specifying
+the company index, role index and prefixes of the fields to be updated.
+
+#### Implementation
+The `editRole` command is primarily implemented by `EditRoleCommandParser` a class that extends `Parser` and `EditRoleCommand` a class that extends `Command`. The `EditRoleCommand` has an inner class `EditRoleDescriptor` that holds the changes to the `Role`.
```
Comment (kerbysoh) [inline docs/DeveloperGuide.md]: Comma should be added between words "Parser" and "and" to avoid confusion

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -105,6 +105,35 @@ The `JsonAdaptedCompany` also contains a list of roles in `List<JsonAdaptedRole>
...
+2. The `EditRoleCommandParser#parse()` method will create an `EditRoleDescriptor` object with all the non-empty fields with the fields' prefixes that are specified by the user such as `s/`,  `b/`, `d/`, etc. In this example, `EditRoleDescriptor#setDescription()` will be used as the description `d/` is a non-empty field with `react js`.
+3. Then the `EditRoleCommandParser#parse()` method will create an `EditRoleCommand` object with the company index `1`, role index `1` and the `EditRoleDescriptor` object.
+4. The `EditRoleCommand` object will be returned to the `LogicManager` and will then invoke the `EditRoleCommand#execute()` method to implement the changes.
+5. The  `EditRoleCommand#execute()` will check the validity of both the indexes, and invoke the `Model#setRole` method.
+6. The  `Model#setRole` with the company index will set the role to be edited with a new role containing the changes. Then the `Model#updateFilteredRoleList()` will update the changes and reflects the changes in the application to the user.
```
Comment (kerbysoh) [inline docs/DeveloperGuide.md]: Model#updateFilteredRoleList() does not update the changes but rather filters the RoleList such that it only displays the edited Role

Comment (kerbysoh) [inline docs/DeveloperGuide.md]: Should Model#setRole be Model#setRole() instead?

**Review by kerbysoh — CHANGES_REQUESTED** [review]
Some minor changes suggested.

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (BrendonLau) [reply docs/DeveloperGuide.md]: yup thanks fo pointing out.

Comment (BrendonLau) [reply docs/DeveloperGuide.md]: Ok got it.

**Pull request merged**